pandas/core/categorical.py

```diff
@@ -361,20 +351,6 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
 
             codes = _get_codes_for_values(values, dtype.categories)
 
-            # TODO: check for old style usage. These warnings should be removes
-            # after 0.18/ in 2016
-            if (is_integer_dtype(values) and
-                    not is_integer_dtype(dtype.categories)):
-                warn("Values and categories have different dtypes. Did you "
```

Comment:
do these get hit anywhere in the tests?

Reply:
fine with removing them. pls add a note in the deprecations removal section.

Reply:
(done)